internal/mesh/internal/types/intermediate/types.go

```diff
@@ -27,41 +28,47 @@ type CombinedDestinationRef struct {
 	ExplicitDestinationsID *pbresource.ID
 }
 
+// Deprecated: types.DecodedServiceEndpoints
 type ServiceEndpoints struct {
 	Resource  *pbresource.Resource
 	Endpoints *pbcatalog.ServiceEndpoints
 }
 
+// Deprecated: types.DecodedService
 type Service struct {
 	Resource *pbresource.Resource
 	Service  *pbcatalog.Service
 }
 
+// Deprecated: types.DecodedDestinations
 type Destinations struct {
 	Resource     *pbresource.Resource
 	Destinations *pbmesh.Upstreams
 }
 
+// Deprecated: types.DecodedWorkload
 type Workload struct {
 	Resource *pbresource.Resource
 	Workload *pbcatalog.Workload
 }
 
+// Deprecated: types.DecodedProxyStateTemplate
 type ProxyStateTemplate struct {
 	Resource *pbresource.Resource
 	Tmpl     *pbmesh.ProxyStateTemplate
 }
 
+// Deprecated: types.DecodedProxyConfiguration
 type ProxyConfiguration struct {
 	Resource *pbresource.Resource
 	Cfg      *pbmesh.ProxyConfiguration
 }
 
 type Destination struct {
-	Explicit         *pbmesh.Upstream
-	ServiceEndpoints *ServiceEndpoints
-	Identities       []*pbresource.Reference
```

Comment:
These fields are related to the cluster generation, which for something with xRoutes may have more than one, hence needing to move this data down into the `ComputedPortRoutes.Targets` field.